packages/extension-vscode/src/utils/problems.ts

```diff
@@ -52,6 +52,7 @@ export const problemToDiagnostic = (problem: Problem, textDocument: TextDocument
     return {
         code: problem.hintId,
         codeDescription: { href: docHref },
+        data: problem,
```

Comment:
Let's wrap `problem` in an object here, just in case we need to attach additional metadata:
``` typescript
...
data: {
  problem
}
...
```